packages/itwin/tree-widget/src/types.ts

```diff
@@ -5,34 +5,46 @@
 
 import type { SpatialContainmentTreeProps, StagePanelLocation, StagePanelSection } from "@itwin/appui-react";
 import type { SelectableContentDefinition } from "@itwin/components-react";
-import type { IModelConnection } from "@itwin/core-frontend";
+import type { IModelConnection, ScreenViewport } from "@itwin/core-frontend";
+import { IPresentationTreeDataProvider } from "@itwin/presentation-components";
 import type { CategoryTreeProps } from "./components/trees/category-tree/CategoriesTree";
 import type { ModelsTreeProps } from "./components/trees/models-tree/ModelsTree";
 
+export interface TreeHeaderButtonProps {
+  viewport?: ScreenViewport;
+  iModel?: IModelConnection;
+}
+
+export interface ModelsTreeHeaderButtonProps extends TreeHeaderButtonProps {
+  availableModels?: string[];
+}
+
+export interface CategoriesTreeHeaderButtonProps extends TreeHeaderButtonProps {
+  filteredProvider?: IPresentationTreeDataProvider;
+}
+
 export interface IModelContentTreeProps
   extends Omit<React.AllHTMLAttributes<HTMLDivElement>, "children"> {
   iModel: IModelConnection;
 }
 
-export type ModelTreeProps = Omit<
-ModelsTreeProps,
+export interface ModelTreeProps extends Omit<ModelsTreeProps,
 | "iModel"
 | "activeView"
 | "width"
 | "height"
 | "filterInfo"
 | "onFilterApplied"
->;
+> { TreeHeaderButtons?: Array<(props: ModelsTreeHeaderButtonProps) => React.ReactNode> }
```

Comment:
Property name should start with lower case letter

Reply:
fixed